after_sign_hook.js

```diff
@@ -1,46 +1,45 @@
-const fs = require('fs');
-const path = require('path');
-var electron_notarize = require('electron-notarize');
-
-module.exports = async function(params) {
-  // Only notarize the app on Mac OS only.
-  if (params.electronPlatformName !== 'darwin') {
-    return;
-  }
-  console.log('afterSign hook triggered', params);
-
-  if (!process.env.CIRCLE_TAG || process.env.CIRCLE_TAG.length === 0) {
-    console.log('Not on a tag. Skipping notarization');
-    return;
-  }
-
-  // Same appId in electron-builder.
-  let appId = 'com.automattic.wordpress';
-
-  let appPath = params.appOutDir
-    ? path.join(
-        params.appOutDir,
-        `${params.packager.appInfo.productFilename}.app`
-      )
-    : params.artifactPaths[0].replace(new RegExp('.blockmap'), '');
-
-  if (!fs.existsSync(appPath)) {
-    throw new Error(`Cannot find application at: ${appPath}`);
-  }
-
-  console.log(`Notarizing ${appId} found at ${appPath}`);
-
-  try {
-    await electron_notarize.notarize({
-      appBundleId: appId,
-      appPath: appPath,
-      appleId: process.env.NOTARIZATION_ID,
-      appleIdPassword: process.env.NOTARIZATION_PWD,
-      ascProvider: 'AutomatticInc',
-    });
-  } catch (error) {
-    console.error(error);
-  }
-
-  console.log(`Done notarizing ${appId}`);
-};
+#!/usr/bin/env node
+
+const path = require( 'path' );
+const { notarize } = require( 'electron-notarize' );
+
+const APP_ID = 'com.automattic.wordpress';
+const NOTARIZATION_ASC_PROVIDER = 'AutomatticInc';
+const NOTARIZATION_ID = process.env.NOTARIZATION_ID;
+const NOTARIZATION_PWD = process.env.NOTARIZATION_PWD;
+
+const shouldNotarize = process.platform === 'darwin' && !!process.env.CIRCLE_TAG;
+
+function elapsed( start ) {
+	const now = new Date();
+
+	const ms = Math.abs( now.getTime() - start.getTime() );
+	const diff = new Date( ms );
+
+	return `${ diff.getMinutes() } minutes, ${ diff.getSeconds() } seconds`;
+}
+
+module.exports = async function( context ) {
+	if ( ! shouldNotarize ) {
+		return;
+	}
+
+	try {
+		const app = path.join( context.appOutDir, `${ context.packager.appInfo.productFilename }.app` );
+		const appName = path.basename( app );
+
+		const start = new Date();
+		console.log( `  • notarizing ${ appName }...` );
+		await notarize( {
+			appBundleId: APP_ID,
+			appPath: app,
+			appleId: NOTARIZATION_ID,
+			appleIdPassword: NOTARIZATION_PWD,
+			ascProvider: NOTARIZATION_ASC_PROVIDER,
+		} )
+		console.log( `  • done notarizing ${ appName }, took ${ elapsed( start ) }` );
```

Comment:
this `elapsed()` may be more complicated than we need it to be.

do we expect it to take many minutes? would "took 2590s" be too unhelpful?
could also consider using a common idiom, `humanTime()` or `humanDiff()`
while keeping the rest easy so that we're not mingling elapsed time and display

```js
const start = Date.now(); // returns seconds
…
const finish = Date.now();
console.log( ` • done notarizing ${ appName }, took ${ ( finish - start ) / 1000 } seconds` );
// or
console.log( ` • done notarizing ${ appName }, took ${ humanTime( finish - start ) } seconds` );

/**
 * We don't expect to take hours
 */
function humanTime( ms ) {
  // convert to time string
}
```

Probably I'm guessing that the human time isn't necessary, that seconds are probably enough - what are they used for? For instance, my shell already tells me how long the last command took if it takes more than 10 seconds; won't CircleCI report runtime as well?

In summary:
 - it seems like we're going out of our way to provide this
 - `elapsed` seems to cross both abstractions of time and display (what does it mean if we call `diff.getYear()` and it returns `69`?) when it could instead focus on just display while keeping the time element simpler, e.g. by using `Date.now()` to return seconds

Reply:
> do we expect it to take many minutes? would "took 2590s" be too unhelpful?

and

> won't CircleCI report runtime as well?

Notarization takes anywhere in the ballpark of ~5 minutes to an hour ( per Apple docs), and I have seen this metric vary significantly within that range. Circle CI provides the total runtime of a given job, but doesn't provide info about the time when console output within a job is updated. Outputting this info as minutes-and-seconds is immensely helpful and makes it more immediate/intuitive to grok, so I'm inclined to keep the format as-is.

> it seems like we're going out of our way to provide this

The turnaround of Apple's servers is useful to know on an ongoing basis, and tracking this stat will be informative in evaluating any significant deviation from the norm.

Reply:
> Outputting this info as minutes-and-seconds is immensely helpful and makes it more immediate/intuitive to grok, so I'm inclined to keep the format as-is.

👍 

> The turnaround of Apple's servers is useful to know on an ongoing basis, and tracking this stat will be informative in evaluating any significant deviation from the norm.

Should this be reported in computer-readable format somewhere? Fire off a stat automattically?

Reply:
> Should this be reported in computer-readable format somewhere? Fire off a stat automattically?

Yeah that's a good thought, and might make this more useful down the road. I'd certainly like to keep that in mind for a potential future enhancement. For the time being we can manually look up the build logs for any given tag (since we only notarize on tag) and make an assessment that way (if and when we need do).